rust-runtime/aws-smithy-runtime-api/src/client/retries/classifiers.rs

```diff
@@ -238,33 +279,196 @@ impl ClassifyRetry for SharedRetryClassifier {
     }
 }
 
-impl ValidateConfig for SharedRetryClassifier {}
+impl ValidateConfig for SharedRetryClassifier {
+    fn validate_final_config(
+        &self,
+        runtime_components: &RuntimeComponents,
+        _cfg: &ConfigBag,
+    ) -> Result<(), BoxError> {
+        let retry_classifiers = runtime_components.retry_classifiers_slice();
+        let out_of_order: Vec<_> = retry_classifiers
+            .windows(2)
+            .filter(|&w| w[0].value().priority() > w[1].value().priority())
+            .collect();
+
+        if out_of_order.is_empty() {
+            Ok(())
+        } else {
+            Err("retry classifiers are mis-ordered; this is a bug".into())
+        }
```

Comment:
Should we only run this check in debug builds?

Reply:
We don't conditionally enable validation for the other components. Why should this one be different?